src/LtiAssignmentsGradesService.php

```diff
@@ -14,24 +14,45 @@ public function getScope(): array
         return $this->getServiceData()['scope'];
     }
 
+    // https://www.imsglobal.org/spec/lti-ags/v2p0#assignment-and-grade-service-claim
+    // When an LTI message is launching a resource associated to one and only one lineitem,
```

Comment:
What is an LTI message? Looking at the link you added but still can't get it

Reply:
An Message is essentially the contents of an LTI launch. This probably isn't 100% accurate, but usually when I see "LTI message", I translate that to "LTI launch" in my head.

Reply:
Yeah, I just copied and pasted that from the docs -- the terminology is definitely confusing, but a launch can contain a line item. For instance, when canvas launches a deep link from an assignment, the line item for that assignment will already be defined and included in the launch (in my experience).